packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -74,6 +74,243 @@ class PlatformViewsService {
+  ///
+  /// See Android's [MotionEvent.PointerCoords#x](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#x).
+  final double x;
+
+  /// The Y component of the pointer movement.
+  ///
+  /// See Android's [MotionEvent.PointerCoords#y](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#y).
+  final double y;
+
+  List<double> _asList() {
+    return <double>[
+      orientation,
+      pressure,
+      size,
+      toolMajor,
+      toolMinor,
+      touchMajor,
+      touchMinor,
+      x,
+      y,
+    ];
+  }
+
+  @override
+  String toString() {
+    return 'AndroidPointerCoords(orientation: $orientation, pressure: $pressure, size: $size, toolMajor: $toolMajor, toolMinor: $toolMinor, touchMajor: $touchMajor, touchMinor: $touchMinor, x: $x, y: $y)';
+  }
+}
+
+/// A Dart version of Android's [MotionEvent](https://developer.android.com/reference/android/view/MotionEvent).
+class AndroidMotionEvent {
+  /// Creates an AndroidMotionEvent.
+  ///
+  /// All parameters must not be null.
+  const AndroidMotionEvent({
+    @required this.downTime,
+    @required this.eventTime,
+    @required this.action,
+    @required this.pointerCount,
+    @required this.pointerProperties,
+    @required this.pointerCoords,
+    @required this.metaState,
+    @required this.buttonState,
+    @required this.xPrecision,
+    @required this.yPrecision,
+    @required this.deviceId,
+    @required this.edgeFlags,
+    @required this.source,
+    @required this.flags
+  }) : assert(downTime != null),
+       assert(eventTime != null),
+       assert(action != null),
+       assert(pointerCount != null),
+       assert(pointerProperties != null),
+       assert(pointerCoords != null),
+       assert(metaState != null),
+       assert(buttonState != null),
+       assert(xPrecision != null),
+       assert(yPrecision != null),
+       assert(deviceId != null),
+       assert(edgeFlags != null),
+       assert(source != null),
+       assert(flags != null);
+
+  /// The time (in ms) when the user originally pressed down to start a stream of position events,
+  /// relative to an arbitrary timeline.
+  ///
+  /// See Android's [MotionEvent#getDownTime](https://developer.android.com/reference/android/view/MotionEvent.html#getDownTime()).
+  final int downTime;
+
+  /// The time this event occurred, relative to an arbitrary timeline.
+  ///
+  /// See Android's [MotionEvent#getEventTime](https://developer.android.com/reference/android/view/MotionEvent.html#getEventTime()).
+  final int eventTime;
+
+  /// A value representing the kind of action being performed.
+  ///
+  /// See Android's [MotionEvent#getAction](https://developer.android.com/reference/android/view/MotionEvent.html#getAction()).
+  final int action;
+
+  /// The number of pointers that are part of this event.
+  /// This must be equivalent to the length of `pointerProperties` and `pointerCoords`.
```

Comment:
if it must, could we just get it from those lists instead of requiring it?

Reply:
I'd rather to keep an exact mirror of the Android API, in case we'll find out in the future some exception when it actually makes sense to pass different numbers. This "must" is to my best understanding a of today.